### ISSUE TITLE
arrayofstruct  Linter is not aware of `AtLeastOneOf` or `ExactlyOneOf` markers

### ISSUE THREAD

**Issue by snorwin** [kubernetes-sigs/kube-api-linter#178]
Currently, the linter enforces that at least one field in a struct must have a required  marker (e.g., `+required`). However, it should also accept cases where the entire struct includes a validation marker that has an equivalent  meaning, e.g.:
```
+kubebuilder:validation:AtLeastOneOf
+kubebuilder:validation:ExactlyOneOf
```
In such cases, the struct-level validation marker already enforces that at least one field is required, so the linter should not raise an error.

It might make sense to add a configuration option to this linter that allows specifying field- and type-level markers which should be interpreted as fulfilling the required constraint.

**Comment by snorwin** [comment]
@JoelSpeed thanks for checking this. For our use case, it would be helpful if `ExactlyOneOf` could also be considered by the arrayofstruct linter.

I did some further testing and noticed that the linter doesn’t seem to detect required fields in embedded structs. This might be an edge case, and I’m not sure how straightforward it would be to address, but I thought it is worth mentioning.
